sqlalchemy_dremio/base.py

```diff
@@ -0,0 +1,266 @@
+        if seq_col is column:
+            colspec += " AUTOINCREMENT"
+        else:
+            colspec += " " + self.dialect.type_compiler.process(column.type)
+
+            if column.nullable is not None and not column.primary_key:
+                if not column.nullable or column.primary_key:
+                    colspec += " NOT NULL"
+                else:
+                    colspec += " NULL"
+
+            default = self.get_column_default_string(column)
+            if default is not None:
+                colspec += " DEFAULT " + default
+
+        return colspec
+
+    def visit_drop_index(self, drop):
+        index = drop.element
+        self.append("\nDROP INDEX [%s].[%s]" % \
+                        (index.table.name,
+                        self._index_identifier(index.name)))
+
+class DremioIdentifierPreparer(compiler.IdentifierPreparer):
+    reserved_words = compiler.RESERVED_WORDS.copy()
+    reserved_words.update(['value', 'text'])
+    def __init__(self, dialect):
+        super(DremioIdentifierPreparer, self).\
+                __init__(dialect, initial_quote='[', final_quote=']')
+
+
+
+class DremioDialect(default.DefaultDialect):
+    colspecs = {
+        types.Unicode: AcUnicode,
+        types.Integer: AcInteger,
+        types.SmallInteger: AcSmallInteger,
+        types.Numeric: AcNumeric,
+        types.Float: AcFloat,
+        types.DateTime: AcDateTime,
+        types.Date: AcDate,
+        types.String: AcString,
+        types.LargeBinary: AcBinary,
+        types.Boolean: AcBoolean,
+        types.Text: AcText,
+        types.CHAR: AcChar,
+        types.TIMESTAMP: AcTimeStamp,
+    }
+    name = 'dremio'
+    supports_sane_rowcount = False
+    supports_sane_multi_rowcount = False
+
+    poolclass = pool.SingletonThreadPool
+    statement_compiler = DremioCompiler
+    ddl_compiler = DremioDDLCompiler
+    preparer = DremioIdentifierPreparer
+    execution_ctx_cls = DremioExecutionContext
+
+    @classmethod
+    def dbapi(cls):
+        import pyodbc as module
+        return module
+
+    def create_connect_args(self, url):
+        opts = url.translate_connect_args()
+        connectors = ["Driver={Dremio Driver (*.mdb)}"]
+        connectors.append("Dbq=%s" % opts["database"])
+        user = opts.get("username", None)
+        if user:
+            connectors.append("UID=%s" % user)
+            connectors.append("PWD=%s" % opts.get("password", ""))
+        return [[";".join(connectors)], {}]
+
+    def last_inserted_ids(self):
+        return self.context.last_inserted_ids
+
+
+    def has_table(self, connection, tablename, schema=None):
+        result = connection.scalar(
+                        sql.text(
+                            "select count(*) from msysobjects where "
+                            "type=1 and name=:name"), name=tablename
+                        )
+        return bool(result)
+
+    @reflection.cache
+    def get_table_names(self, connection, schema=None, **kw):
```

Comment:
Could we replace with corresponding functions for Dremio?
Perhaps start with what the Drill SA dialect is doing (https://github.com/JohnOmernik/sqlalchemy-drill/blob/master/sqlalchemy_drill/sadrill.py)
and tweak it based on more specific Dremio SQL syntax for introspection and info schemas
https://github.com/dremio/dremio-oss/blob/master/sabot/kernel/src/test/java/com/dremio/exec/sql/TestInfoSchema.java